server/route.go

```diff
@@ -659,12 +659,13 @@ func (s *Server) routeAcceptLoop(ch chan struct{}) {
 	}
 
 	hp := net.JoinHostPort(opts.Cluster.Host, strconv.Itoa(port))
-	s.Noticef("Listening for route connections on %s", hp)
 	l, e := net.Listen("tcp", hp)
 	if e != nil {
-		s.Fatalf("Error listening on router port: %d - %v", opts.Cluster.Port, e)
+		s.Fatalf("Error listening on router port: %d - %v", l.Addr().(*net.TCPAddr).Port, e)
```

Comment:
No, since there is an error, `l` will probably be nil. Just leave it to `opts.Cluster.Port` here, or `port`.